python/lsst/daf/butler/core/config.py

```diff
@@ -623,8 +621,9 @@ def __getitem__(self, name):
         # all further cleverness.
         found_directly = False
         try:
-            data = self._data[name]
-            found_directly = True
+            if isinstance(name, str):
```

Comment:
It does seem painful to have to do an `isinstance` check even though the dict lookup would fail if the item wasn't a string but that's the price we pay for mypy. Especially when this short circuit was there as an optimization to let the fast thing be fast.